File: 3183/3183.py
class Solution:
    def the_number_of_ways_to_make_the_sum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.the_number_of_ways_to_make_the_sum(root.left), self.the_number_of_ways_to_make_the_sum(root.right))

File: 3183/3183.cpp
class Solution {
public:
    int theNumberOfWaysToMakeTheSum(TreeNode* root) {
        if(!root) return 0;
        return 1+max(theNumberOfWaysToMakeTheSum(root->left),theNumberOfWaysToMakeTheSum(root->right));
    }
};